scripts/docs_config/introduction.rst

```diff
@@ -158,5 +129,54 @@ When a client requests an IPC handle for a memory allocated by UMF, UMF does the
 Not every memory provider can and must support IPC operations. It is up to the memory provider implementation to decide if it supports IPC operations. 
 If the corresponding memory provider does not support IPC operations, UMF will return an error when a client requests an IPC handle for a memory object allocated by this memory provider.
 
+==============
+ Fundamentals
+==============
+
+The following section provides fundamentals of the project. More detailed information can be found in
+contributing guide and detailed API pages.
+
+The contribution guide can be found in the repository, in file: `CONTRIBUTING.md`_.
+
+Logging
+============
+
+Logging in UMF is handled by logger. There are several levels of logging: *debug*, *info*, *warning*, and *error*.
+The level of logging determines what messages will be printed, ie. the level set to *warning* means all messages at levels *warning* and *error* will be printed.
+
+By default, there is a guarantee that *error* messages are flushed immediately. One can change this behavior to flush on lower-level messages.
+
+Loggers redirect messages to *stdout*, *stderr*, or a file.
+
+By default, only fatal messages are printed. To enable logger you have to set **UMF_LOG** environment variable which have following syntax for setting logger options::
+
+    UMF_LOG="[level:debug|info|warning|error|fatal];[flush:debug|info|warning|error|fatal];[output:stdout|stderr|file,<path>];[timestamp:yes|no];[pid:yes|no]"
+
+* level - a log level, meaning that only messages from this level and above are printed. Possible values, from the lowest level to the highest one: *debug*, *info*, *warning*, *error*, *fatal*.
+
+* flush - a flush level, meaning that messages at this level and above are guaranteed to be flushed immediately, possible values are the same as above.
+
+* output - indicates where messages should be printed. Possible values are: *stdout*, *stderr* and *file*, when providing a *file* output option, a *<path>* is required.
+
+* timestamp - add a timestamp to log message.
+
+* pid - add thread and process ids to log message.
+
+.. note::
+  For output to file, a path to the file has to be provided after a comma, like in the example above. The path has to exist, file will be created if not existing. Path to the file should be no longer than 255 characters long.
+  The output parameter is required, all other logger options are optional. The defaults are set when options are not provided in the environment variable.
+  Options have to be separated with `;`, option names and their values with `:`. Additionally, when providing *file* output, the keyword *file* and a path to a file
```

Comment:
It's hard to read the separators, maybe enclose them with quotation marks (`";"`) or write their names and put characters in brackets, like `a semicolon (;)`?

Reply:
Good catch, done.